jsonrpc/eth_endpoint.go

```diff
@@ -111,28 +111,44 @@ func (e *Eth) SendTransaction(arg *txnArgs) (interface{}, error) {
 	return transaction.Hash.String(), nil
 }
 
-// GetTransactionByHash returns a transaction by his hash
+// GetTransactionByHash returns a transaction by its hash.
+// If the transaction is still pending -> return the txn with some fields omitted
+// If the transaction is sealed into a block -> return the whole txn with all fields
 func (e *Eth) GetTransactionByHash(hash types.Hash) (interface{}, error) {
```

Comment:
This method never returns an error. If this is correct behaviour, remove the error from returned values

Reply:
I believe all of our endpoint methods need to return a `interface{}, error` response:
https://github.com/umbracle/go-web3/blob/e590eac2e9fd8e635782af2b3b9dbb49117dd003/jsonrpc/transport/http.go#L67-L69

Reply:
I see now, all good